gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapperIT.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.hl7.fhir.dstu3.model.AllergyIntolerance;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import lombok.SneakyThrows;
+
+@SpringBootTest
+@ExtendWith({SpringExtension.class, MockitoExtension.class})
+public class ConditionMapperIT {
+
+    private static final String CONDITION_RESOURCES_BASE = "xml/condition/";
+    private static final String NAMED_STATEMENT_REF_ID = "NAMED_STATEMENT_REF_ID";
+    private static final String PATIENT_ID = "PATIENT_ID";
+    private static final String PRACTISE_CODE = "TESTPRACTISECODE";
+
+    private static final String SNOMED_SYSTEM = "http://snomed.info/sct";
+    private static final String OBSERVATION_CONCEPT_ID = "240534005";
+    private static final String OBSERVATION_CONCEPT_DESCRIPTION = "Hand wart";
+    private static final String ALLERGY_CONCEPT_ID = "442728009";
+    private static final String ALLERGY_CONCEPT_DESCRIPTION = "Non-coeliac gluten sensitivity";
+
+    @Autowired
+    private ConditionMapper conditionMapper;
+
+    private Patient patient;
+
+    @BeforeEach
+    public void setUp() {
+        patient = (Patient) new Patient().setId(PATIENT_ID);
+    }
+
+    public static Stream<Arguments> ehrExtractsWithObservationActualProblemCode() {
+        return Stream.of(
+            Arguments.of("linkset_problem_observation-in-category.xml"),
+            Arguments.of("linkset-problem-observation-flat.xml"),
+            Arguments.of("linkset-problem-observation-in-topic.xml")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("ehrExtractsWithObservationActualProblemCode")
+    public void When_AddReferences_With_ObservationAsProblem_Expect_CodeAddedToCondition(String filename) {
+
+        final RCMRMT030101UK04EhrExtract ehrExtract = unmarshallEhrExtract(filename);
+        final List<Condition> conditions = conditionMapper.mapResources(ehrExtract, patient, List.of(), PRACTISE_CODE);
+
+        conditionMapper.addReferences(buildBundleWithNamedStatementObservation(), conditions, ehrExtract);
+
+        var code = conditions.get(0).getCode().getCodingFirstRep();
+
+        assertThat(code.getDisplay()).isEqualTo(OBSERVATION_CONCEPT_DESCRIPTION);
+        assertThat(code.getCode()).isEqualTo(OBSERVATION_CONCEPT_ID);
```

Comment:
It might be a matter of personal preference but somehow I found "assertEquals(exp, actValue)" more concise and better to read hence I would suggest using that.

Reply:
I've updated it to use the JUnit method you suggest. I know it's more verbose, but I like the assertJ `isEqualTo` because it states in the method names where the actual and expected values go. I find the ordering in `assertEquals` a little unintuitive. My brain wants to write `assertEquals(actual, expected)`